internal/go_repository.bzl

```diff
@@ -235,6 +234,14 @@ go_repository = repository_rule(
                 "off",
             ],
         ),
+        "build_naming_convention": attr.string(
+            values = [
```

Comment:
GitHub wouldn't let me reply to the previous comment, so I'll reply here:

> Do we have to hard launch this, instead of making it opt-in?

We don't have to, but I'd argue that we should.

In general, I'd prefer not to give people too many configuration options. Each option adds complexity, both in usage and in implementation. In situations where we can figure out what the Right Thing is, we should do that. If we can't infer the Right Thing, or if there are several reasonable Right Things, only then should we ask the user.

I definitely think the new naming convention is the Right Thing; `go_default_library` is a historical accident. We need to support the old convention because people are still using it, but the new one should really be the default going forward.